lib/r10k/action/puppetfile/purge.rb

```diff
@@ -31,6 +16,12 @@ def call
           logger.error R10K::Errors::Formatting.format_exception(e, @trace)
           false
         end
+
+        private
+
+        def allowed_initialize_opts
+          super.merge(root: :self, puppetfile: :self, moduledir: :self)
```

Comment:
I'll centralize around `:self`, and at some point I should yank :true.

In the future I should think very seriously about dropping 1.9 support so that we can use Ruby keyword args instead of this workaround.